core/src/main/java/org/apache/accumulo/core/conf/AccumuloConfiguration.java

```diff
@@ -435,6 +438,60 @@ Integer getDeprecatedScanThreads(String name) {
     return null;
   }
 
+  private static class RefCount<T> {
+    T obj;
+    long count;
+
+    RefCount(long c, T r) {
+      this.count = c;
+      this.obj = r;
+    }
+  }
+
+  private class Deriver<T> implements Supplier<T> {
+
+    private AtomicReference<RefCount<T>> refref;
+    private Function<AccumuloConfiguration,T> factory;
+
+    Deriver(Function<AccumuloConfiguration,T> factory) {
+      this.factory = factory;
+      refref = new AtomicReference<>();
+    }
+
+    @Override
+    public T get() {
+      RefCount<T> rc = refref.get();
+
+      long uc = getUpdateCount();
+
+      if (rc == null || rc.count != uc) {
+        T newObj = factory.apply(AccumuloConfiguration.this);
+
+        RefCount<T> nrc = new RefCount<>(uc, newObj);
+        refref.compareAndSet(rc, nrc);
```

Comment:
Yeah I thought about doing a loop but decided against this because no matter what you do, after this functions returns the object could immediately become stale.  So I concluded the cycles would be needlessly spent trying to accomplish a goal that can never be accomplished.  I will add a comment about why the return value is ignored and a loop was not used.